python/code_challenges/stack_queue_animal_shelter.py

```diff
@@ -1,13 +1,48 @@
+from distutils.log import error
 from data_structures.queue import Queue
 
 
 class AnimalShelter:
-    pass
 
+    def __init__(self):
+        self.dog_queue = Queue()
+        self.cat_queue = Queue()
+
+    def enqueue(self, animal):
+        if animal.pref == "cat":
+
+            self.cat_queue.enqueue(animal)
+
+        if animal.pref == "dog":
+
+            self.dog_queue.enqueue(animal)
+
+    def dequeue(self, pref):
+        if pref == "cat":
+
+            return self.cat_queue.dequeue()
+
+        if pref == "dog":
+
+            return self.dog_queue.dequeue()
+
+        else:
+            return None
+
+class Animal():
+
+    def __init__(self, name=None):
+        self.name = name
+
+class Dog(Animal):
+
+    def __init__(self, name=None, pref="dog"):
+        self.pref = pref
+
+class Cat(Animal):
+
+    def __init__(self, name=None, pref="cat"):
+        self.pref = pref
```

Comment:
I'm pretty sure I see what you're trying to do here, but it's worth noting that `__init__` from Dog and Cat are overriding the Animal `__init__`. Because of this, name isn't actually being assigned in your Dog and Cat objects. If you wanted to stick with this approach, you can use `super()` to access parent class methods that would normally be overridden, like this:
```Python
class Dog(Animal):
    def __init__(self, name=None, pref="dog"):
        super().__init__(name)
        self.pref = pref
```